src/intro_movie_scenes_lib.c

```diff
@@ -0,0 +1,528 @@
+    SetVec(template->position, 0, 0, 0);
+    SetVec(template->scale, FX32_ONE, FX32_ONE, FX32_ONE);
+    template->rotation = 0;
+    template->priority = 0;
+    template->whichScreen = whichScreen;
+    template->heapId = HEAP_ID_INTRO_MOVIE;
+}
+
+void IntroMovie_RendererSetSurfaceCoords(IntroMovieOvyData *data, int mainx, int mainy, int subx, int suby) {
+    G2dRenderer_SetSubSurfaceCoords(&data->spriteRenderer, subx * FX32_ONE, suby * FX32_ONE);
+    G2dRenderer_SetMainSurfaceCoords(&data->spriteRenderer, mainx * FX32_ONE, mainy * FX32_ONE);
+}
+
+void IntroMovie_StartBlendFadeEffect(IntroMovieBgBlendAnim *data, int plane1, int plane2, u8 duration, int direction, int screen) {
+    data->counter = 0;
+    data->ev = 0;
+    data->finished = 0;
+    data->topScreen = screen;
+    data->plane1 = plane1;
+    data->plane2 = plane2;
+    data->duration = duration;
+    data->direction = direction;
+    data->task = SysTask_CreateOnMainQueue(Task_IntroMovie_BlendFadeEffect, data, 0);
+}
+
+void Task_IntroMovie_BlendFadeEffect(SysTask *task, void *pVoid) {
+    IntroMovieBgBlendAnim *data = (IntroMovieBgBlendAnim *)pVoid;
+
+    ++data->counter;
+    data->ev = data->counter * 31 / data->duration;
+    if (data->ev >= 31) {
+        data->ev = 31;
+        SysTask_Destroy(data->task);
+        data->task = NULL;
+        data->finished = 1;
+    }
+    int ev;
+    if (data->direction == 0) {
+        ev = data->ev;
+    } else {
+        ev = 31 - data->ev;
+    }
+    if (data->topScreen) {
+        G2_SetBlendAlpha(data->plane1, data->plane2, ev, 31 - ev);
+    } else {
+        G2S_SetBlendAlpha(data->plane1, data->plane2, ev, 31 - ev);
+    }
+}
+
+void IntroMovie_StartBgScroll_VBlank(BgConfig *bgConfig, IntroMovieBgScrollAnim *data, enum GFBgLayer bgId, s16 xChange, s16 yChange, int duration) {
+    IntroMovieBgScrollAnim *obj = &data[IntroMovie_BgLayerToScrollEffectSlot(bgId)];
+    if (obj->active) {
+        GF_ASSERT(FALSE);
+        return;
+    }
+
+    if (duration == 0) {
+        int x = xChange + Bg_GetXpos(bgConfig, bgId);
+        int y = yChange + Bg_GetYpos(bgConfig, bgId);
+        BgSetPosTextAndCommit(bgConfig, bgId, BG_POS_OP_SET_X, x);
+        BgSetPosTextAndCommit(bgConfig, bgId, BG_POS_OP_SET_Y, y);
+        return;
+    }
+
+    obj->bgConfig = bgConfig;
+    obj->bgId = bgId;
+    obj->counter = 0;
+    obj->finished = FALSE;
+    obj->duration = duration;
+    obj->xOrig = Bg_GetXpos(bgConfig, bgId);
+    obj->yOrig = Bg_GetYpos(bgConfig, bgId);
+    obj->xChange = xChange;
+    obj->yChange = yChange;
+    obj->active = TRUE;
+    obj->task = SysTask_CreateOnVBlankQueue(Task_IntroMovie_BgScroll_VBlank, obj, 0);
+}
+
+void IntroMovie_StartBgScroll_NotVBlank(BgConfig *bgConfig, IntroMovieBgScrollAnim *data, enum GFBgLayer bgId, s16 xChange, s16 yChange, int duration) {
+    IntroMovieBgScrollAnim *obj = &data[IntroMovie_BgLayerToScrollEffectSlot(bgId)];
+    if (obj->active) {
+        GF_ASSERT(FALSE);
```

Comment:
also worth thinking about a GF_ASSERT_RETURN cause this is very common

Reply:
disagree

Reply:
fair, I can see this is not a very popular opinion